tests/util/skvbc_history_tracker.py

```diff
@@ -724,3 +766,106 @@ def _get_matching_request(self, rpy):
         causal_state = self.outstanding[(rpy.client_id, rpy.seq_num)]
         index = causal_state.req_index
         return (self.history[index], index)
+
+    async def verify_linearizability(self):
+         try:
+             # Use a new client, since other clients may not be responsive due to
+             # past failed responses.
+             client = await self.skvbc.bft_network.new_client()
+             last_block_id = await self.get_last_block_id(client)
+             print(f'Last Block ID = {last_block_id}')
+             missing_block_ids = self.get_missing_blocks(last_block_id)
+             print(f'Missing Block IDs = {missing_block_ids}')
+             blocks = await self.get_blocks(client, missing_block_ids)
+             self.fill_missing_blocks(blocks)
+             self.verify()
+         except Exception as e:
+             print(f'retries = {client.retries}')
+             self.status.end_time = time.monotonic()
+             print("HISTORY...")
+             for i, entry in enumerate(self.history):
+                 print(f'Index = {i}: {entry}\n')
+             print("BLOCKS...")
+             print(f'{self.blocks}\n')
+             print(str(self.status), flush=True)
+             print("FAILURE...")
+             raise(e)
+
+    async def get_blocks(self, client, block_ids):
+        blocks = {}
+        for block_id in block_ids:
+            retries = 12 # 60 seconds
+            for i in range(0, retries):
+                try:
+                    msg = kvbc.SimpleKVBCProtocol.get_block_data_req(block_id)
+                    blocks[block_id] = kvbc.SimpleKVBCProtocol.parse_reply(await client.read(msg))
+                    break
+                except trio.TooSlowError:
+                    if i == retries - 1:
+                        raise
+            print(f'Retrieved block {block_id}')
+        return blocks
+
+    async def get_last_block_id(self, client):
+        msg = kvbc.SimpleKVBCProtocol.get_last_block_req()
+        return kvbc.SimpleKVBCProtocol.parse_reply(await client.read(msg))
+
+    async def run_concurrent_ops(self, num_ops):
```

Comment:
This is a single specific test case. I don't think it should be part of the tracker without some slight modification. For example, the percentage of reads and writes is hardcoded (.7) and the set size is limited (to try to cause conflicts), and also the max concurrency is limited. We may want to make these configurable in a future PR.

Reply:
+1, makes sense, let's return `run_concurrent_ops()` in the test module. We can keep it there until we figure it can be useful in other tests. Then we can generalize it and make it a utility, rather than a test method,